components/style/traversal.rs

```diff
@@ -62,6 +62,23 @@ impl LogBehavior {
     fn allow(&self) -> bool { matches!(*self, MayLog) }
 }
 
+/// The kind of traversals we could perform.
+#[derive(Debug, Copy, Clone)]
+pub enum TraversalKind {
```

Comment:
Nit: I think it would be a little more specific to refer to this as TraversalDriver. It's a bit of a judgement call though so I won't insist.